src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -372,4 +373,145 @@ public void deleteInventory(MedicalInventory medicalInventory) throws OHServiceE
+	 * @return {@link MedicalInventory}. It could be {@code null}.
+	 * @throws OHDataValidationException
+	 */
+	public MedicalInventory actualizeMedicalInventoryRow(MedicalInventory inventory) throws OHServiceException {
+		LocalDateTime movFrom = inventory.getInventoryDate();
+		LocalDateTime movTo = TimeTools.getNow();
+		// Fetch all the inventory row of that inventory
+		int id = inventory.getId();
+		List<MedicalInventoryRow> inventoryRowList = medicalInventoryRowManager.getMedicalInventoryRowByInventoryId(id);
+		// TODO: To decide if to make allMedicals parameter
+		boolean allMedicals = true;
+		List<Movement> movs = new ArrayList<>();
+		List<Medical> inventoryMedicalsList = inventoryRowList.stream()
+						.map(MedicalInventoryRow::getMedical)
+						.distinct()
+						.collect(Collectors.toList());
+		if (allMedicals) {
+			// Fetch all movements without filtering by medical code
+			movs.addAll(movBrowserManager.getMovements(null, null, null, null, movFrom, movTo, null, null, null, null));
+		} else {
+			// Fetch only movements concerning inventoryRowSearchList list
+			for (Medical medical : inventoryMedicalsList) {
+				movs.addAll(movBrowserManager.getMovements(medical.getCode(), null, null, null, movFrom, movTo, null, null, null, null));
+			}
+		}
+		// Get all the lot of the movements
+		List<Lot> lotOfMovements = movs.stream().map(m -> m.getLot()).collect(Collectors.toList());
+		// Remove duplicates by converting the list to a set
+		Set<Lot> uniqueLots = new HashSet<>(lotOfMovements);
+		// Convert the set back to a list
+		List<Lot> uniqueLotList = new ArrayList<>(uniqueLots);
+		// Cycle fetched movements to see if they impact inventoryRowSearchList
+		for (Lot lot : uniqueLotList) {
+			String lotCodeOfMovement = lot.getCode();
+			Medical medical = lot.getMedical();
+			Integer medicalCode = medical.getCode();
+			// Fetch also empty lots because some movements may have discharged them completely
+			Optional<Lot> optLot = movStockInsertingManager.getLotByMedical(medical, false).stream().filter(l -> l.getCode().equals(lotCodeOfMovement))
+							.findFirst();
+			double mainStoreQty = optLot.get().getMainStoreQuantity();
+
+			// Search for the specific Lot and Medical in inventoryRowSearchList (Lot should be enough)
+			Optional<MedicalInventoryRow> matchingRow = inventoryRowList.stream()
+							.filter(row -> row.getLot().getCode().equals(lotCodeOfMovement) && row.getMedical().getCode().equals(medicalCode))
+							.findFirst();
+
+			if (matchingRow.isPresent()) {
+				MedicalInventoryRow medicalInventoryRow = matchingRow.get();
+				double theoQty = medicalInventoryRow.getTheoreticQty();
+				if (mainStoreQty != theoQty) {
+					// Update Lot
+					medicalInventoryRow.setTheoreticQty(mainStoreQty);
+					medicalInventoryRowManager.updateMedicalInventoryRow(medicalInventoryRow);
+				}
+			} else {
+				// TODO: to decide if to give control to the user about this
+				double realQty = mainStoreQty;
```

Comment:
why assign the value to a new variable if it is only used once the following method call?

Reply:
also here, look at the TODO comment on the line above, for future versions